extensions-contrib/opentelemetry-emitter/src/main/java/org/apache/druid/emitter/opentelemetry/OpenTelemetryEmitter.java

```diff
@@ -0,0 +1,120 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.emitter.opentelemetry;
+
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.StatusCode;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import org.apache.druid.java.util.common.logger.Logger;
+import org.apache.druid.java.util.emitter.core.Emitter;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.apache.druid.java.util.emitter.service.ServiceMetricEvent;
+import org.joda.time.DateTime;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+public class OpenTelemetryEmitter implements Emitter
+{
+  private static final Logger log = new Logger(OpenTelemetryEmitter.class);
+  private static final DruidContextTextMapGetter DRUID_CONTEXT_MAP_GETTER = new DruidContextTextMapGetter();
+  private final Tracer tracer;
+  private final TextMapPropagator propagator;
+
+  OpenTelemetryEmitter()
+  {
+    tracer = GlobalOpenTelemetry.getTracer("druid-opentelemetry-extension");
+    propagator = GlobalOpenTelemetry.getPropagators().getTextMapPropagator();
+  }
+
+  @Override
+  public void start()
+  {
+    log.debug("Starting OpenTelemetryEmitter");
+  }
+
+  @Override
+  public void emit(Event e)
+  {
+    if (!(e instanceof ServiceMetricEvent)) {
+      return;
+    }
+    ServiceMetricEvent event = (ServiceMetricEvent) e;
+
+    // We generate spans for the following types of events:
+    // query/time
+    if (!event.getMetric().equals("query/time")) {
+      return;
+    }
+
+    emitQueryTimeEvent(event);
+  }
+
+  private void emitQueryTimeEvent(ServiceMetricEvent event)
+  {
+    Map<String, String> druidContext = getContextAsString(event);
+    Context opentelemetryContext = propagator.extract(Context.current(), druidContext, DRUID_CONTEXT_MAP_GETTER);
+
+    try (Scope scope = opentelemetryContext.makeCurrent()) {
+      DateTime endTime = event.getCreatedTime();
+      DateTime startTime = endTime.minusMillis(event.getValue().intValue());
+
+      Span span = tracer.spanBuilder(event.getService())
+                        .setStartTimestamp(startTime.getMillis(), TimeUnit.MILLISECONDS)
+                        .startSpan();
+      druidContext.forEach(span::setAttribute);
```

Comment:
shouldn't we be excluding the opentelemetry context entries from attributes (e.g. traceparent, baggage, etc.) all the entries that are already extracted by `propagator.extract(..)` and present in the opentelemetry context.

Reply:
> shouldn't we be excluding the opentelemetry context entries from attributes

+1